src/test/java/io/tiledb/spark/TimeTravellingTest.java

```diff
@@ -0,0 +1,246 @@
+    // Test first time interval
+    Dataset<Row> dfRead = load("0", "10");
+    dfRead.createOrReplaceTempView("tmp");
+    List<Row> rows = session().sql("SELECT * FROM tmp").collectAsList();
+
+    // 1st row
+    Row row = rows.get(0);
+    Assert.assertEquals(1, row.getInt(0));
+    Assert.assertEquals(1, row.getInt(1));
+
+    // 2nd row
+    row = rows.get(1);
+    Assert.assertEquals(2, row.getInt(0));
+    Assert.assertEquals(2, row.getInt(1));
+
+    // 3nd row
+    row = rows.get(2);
+    Assert.assertEquals(3, row.getInt(0));
+    Assert.assertEquals(3, row.getInt(1));
+
+    // 4th row
+    row = rows.get(3);
+    Assert.assertEquals(4, row.getInt(0));
+    Assert.assertEquals(4, row.getInt(1));
+
+    // Test second time interval
+    dfRead = load("10", "20");
+    dfRead.createOrReplaceTempView("tmp");
+    rows = session().sql("SELECT * FROM tmp").collectAsList();
+
+    // 1st row
+    row = rows.get(0);
+    Assert.assertEquals(1, row.getInt(0));
+    Assert.assertEquals(101, row.getInt(1));
+
+    // 2nd row
+    row = rows.get(1);
+    Assert.assertEquals(2, row.getInt(0));
+    Assert.assertEquals(102, row.getInt(1));
+
+    // 3nd row
+    row = rows.get(2);
+    Assert.assertEquals(3, row.getInt(0));
+    Assert.assertEquals(103, row.getInt(1));
+
+    // 4th row
+    row = rows.get(3);
+    Assert.assertEquals(4, row.getInt(0));
+    Assert.assertEquals(104, row.getInt(1));
+
+    // Test third time interval
+    dfRead = load("20", "30");
+    dfRead.createOrReplaceTempView("tmp");
+    rows = session().sql("SELECT * FROM tmp").collectAsList();
+
+    // 1st row
+    row = rows.get(0);
+    Assert.assertEquals(1, row.getInt(0));
+    Assert.assertEquals(201, row.getInt(1));
+
+    // 2nd row
+    row = rows.get(1);
+    Assert.assertEquals(2, row.getInt(0));
+    Assert.assertEquals(202, row.getInt(1));
+
+    // 3nd row
+    row = rows.get(2);
+    Assert.assertEquals(3, row.getInt(0));
+    Assert.assertEquals(203, row.getInt(1));
+
+    // 4th row
+    row = rows.get(3);
+    Assert.assertEquals(4, row.getInt(0));
+    Assert.assertEquals(204, row.getInt(1));
+  }
+
+  private Dataset<Row> load(String timestamp_start, String timestamp_end) {
+    return session()
+        .read()
+        .format("io.tiledb.spark")
+        .option("timestamp_start", timestamp_start)
+        .option("timestamp_end", timestamp_end)
+        .option("uri", arrayURI)
+        .load();
+  }
+
+  public void arrayCreate() throws Exception {
+
+    // Create getDimensions
+    Dimension<Integer> dim =
+        new Dimension<Integer>(ctx, "rows", Integer.class, new Pair<Integer, Integer>(1, 4), 2);
+
+    // Create and set getDomain
+    Domain domain = new Domain(ctx);
+    domain.addDimension(dim);
+
+    // Create and add getAttributes
+    Attribute a = new Attribute(ctx, "a", Integer.class);
+
+    ArraySchema schema = new ArraySchema(ctx, TILEDB_DENSE);
+    schema.setTileOrder(TILEDB_ROW_MAJOR);
```

Comment:
If you have time, this could be a good place to try using the parametricity we've discussed in the past in order to run this over all layouts.

Reply:
Will work on this in a different PR that improves testing in the API overall.